x-pack/plugins/reporting/server/browsers/chromium/driver/chromium_driver.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Chrome from 'puppeteer';
+import {
+  ElementPosition,
+  EvalArgs,
+  EvalFn,
+  EvaluateOptions,
+  Logger,
+  ViewZoomWidthHeight,
+} from '../../../../types';
+
+export interface ChromiumDriverOptions {
+  logger: Logger;
+}
+
+const WAIT_FOR_DELAY_MS: number = 100;
+
+export class HeadlessChromiumDriver {
+  public documentNode?: Chrome.JSHandle;
+
+  private readonly page: Chrome.Page;
+  private readonly logger: Logger;
+
+  constructor(page: Chrome.Page, { logger }: ChromiumDriverOptions) {
+    this.page = page;
+    this.logger = logger;
+  }
+
+  public async open(
+    url: string,
+    { headers, waitForSelector }: { headers: Record<string, string>; waitForSelector: string }
+  ) {
+    this.logger.debug(`HeadlessChromiumDriver:opening url ${url}`);
+
+    await this.page.setExtraHTTPHeaders(headers);
+    await this.page.goto(url, { waitUntil: 'networkidle0' });
+
+    this.documentNode = await this.page.evaluateHandle(() => document);
```

Comment:
I only see the `documentNode` property getting assigned, but not used. Looks like that was the case in the previous code too. If that's intended, maybe clarify with a comment?

Reply:
good catch